contracts/contracts/coordination/Coordinator.sol

```diff
@@ -147,14 +147,19 @@ contract Coordinator is AccessControlDefaultAdminRules, FlatRateFeeModel {
         _setRoleAdmin(INITIATOR_ROLE, bytes32(0));
     }
 
-    function setProviderPublicKey(BLS12381.G2Point calldata _publicKey) public {
+    function setProviderPublicKey(BLS12381.G2Point calldata _publicKey) external {
         uint32 lastRitualId = uint32(rituals.length);
-        address provider = application.stakingProviderFromOperator(msg.sender);
+        address stakingProvider = application.stakingProviderFromOperator(msg.sender);
+        require(stakingProvider != address(0), "Operator has no bond with staking provider"); // TODO
 
         ParticipantKey memory newRecord = ParticipantKey(lastRitualId, _publicKey);
-        keysHistory[provider].push(newRecord);
+        // keysHistory[stakingProvider][-1].publicKey != _publicKey; // TODO it's a question
```

Comment:
I'm in favor of adding this check and preventing unnecessary updates. Allowing this update could trigger some confusing behavior (emitting events, other contract calls, etc.). I think it also prevents confusion on the operator's part - Why would they make an unnecessary update? Could be an operator error.

Reply:
agree, but coordinator updates will be in some future PR, here just to not forget